src/api/EscargotPublic.cpp

```diff
@@ -1332,27 +1332,28 @@ bool ObjectRef::defineAccessorProperty(ExecutionStateRef* state, ValueRef* prope
                                            ObjectPropertyName(*toImpl(state), toImpl(propertyName)), ObjectPropertyDescriptor(JSGetterSetter(toImpl(desc.m_getter), toImpl(desc.m_setter)), (ObjectPropertyDescriptor::PresentAttribute)desc.m_attribute));
 }
 
-bool ObjectRef::defineNativeDataAccessorProperty(ExecutionStateRef* state, ValueRef* propertyName, NativeDataAccessorPropertyData* publicData)
+bool ObjectRef::defineNativeDataAccessorProperty(ExecutionStateRef* state, ValueRef* propertyName, NativeDataAccessorPropertyData* publicData, bool actsLikeJSGetterSetter)
 {
-    ObjectPropertyNativeGetterSetterData* innerData = new ObjectPropertyNativeGetterSetterData(publicData->m_isWritable, publicData->m_isEnumerable, publicData->m_isConfigurable, [](ExecutionState& state, Object* self, const EncodedValue& privateDataFromObjectPrivateArea) -> Value {
+    ObjectPropertyNativeGetterSetterData* innerData = new ObjectPropertyNativeGetterSetterData(publicData->m_isWritable, publicData->m_isEnumerable, publicData->m_isConfigurable, [](ExecutionState& state, Object* self, const Value& receiver, const EncodedValue& privateDataFromObjectPrivateArea) -> Value {
         NativeDataAccessorPropertyData* publicData = reinterpret_cast<NativeDataAccessorPropertyData*>(privateDataFromObjectPrivateArea.payload());
-        return toImpl(publicData->m_getter(toRef(&state), toRef(self), publicData));
+        return toImpl(publicData->m_getter(toRef(&state), toRef(self), toRef(receiver), publicData));
     },
                                                                                                nullptr);
 
     if (!publicData->m_isWritable) {
         innerData->m_setter = nullptr;
     } else if (publicData->m_isWritable && !publicData->m_setter) {
-        innerData->m_setter = [](ExecutionState& state, Object* self, EncodedValue& privateDataFromObjectPrivateArea, const Value& setterInputData) -> bool {
+        innerData->m_setter = [](ExecutionState& state, Object* self, const Value& receiver, EncodedValue& privateDataFromObjectPrivateArea, const Value& setterInputData) -> bool {
             return false;
         };
     } else {
-        innerData->m_setter = [](ExecutionState& state, Object* self, EncodedValue& privateDataFromObjectPrivateArea, const Value& setterInputData) -> bool {
+        innerData->m_setter = [](ExecutionState& state, Object* self, const Value& receiver, EncodedValue& privateDataFromObjectPrivateArea, const Value& setterInputData) -> bool {
             NativeDataAccessorPropertyData* publicData = reinterpret_cast<NativeDataAccessorPropertyData*>(privateDataFromObjectPrivateArea.payload());
-            return publicData->m_setter(toRef(&state), toRef(self), publicData, toRef(setterInputData));
+            return publicData->m_setter(toRef(&state), toRef(self), toRef(receiver), publicData, toRef(setterInputData));
         };
     }
 
+    innerData->m_actsLikeJSGetterSetter = actsLikeJSGetterSetter;
```

Comment:
Just a suggestion.
How about setting this member variable(`m_actsLikeJSGetterSetter) through `ObjectPropertyNativeGetterSetterData` constructor?

Reply:
That's good idea!